docs/develop.md

```diff
@@ -42,6 +45,14 @@ Please install *goimports* and configure your source code editor to automaticall
  list the files whose formatting does not conform to *goimports* formatting, and then fix the 
  formatting of all go source files in the repository.  
 
+## Dependency management
+Emmy uses [dep](https://github.com/golang/dep) for dependency management. All direct dependencies 
+and their versions are specified in `Gopkg.toml`. At the moment, minor range locking is used (i.e. 
+we only allow semantic version patches. A quick read of `dep`'s [documentation](https://golang.github.io/dep/docs/daily-dep.html) 
+is recommended to become familiar with the tool. Note that for now, not all direct 
+dependencies are strictly version-locked (some of them have yet to create a release with necessary 
+features that are already implemented in the master branch).
+
```

Comment:
Since this document is a guide for developing & contributing to emmy, would you mind also adding some indication that `Gopkg.toml` and `Gopkg.lock` should both be versioned with the rest of the repository? 

I think we should make it clear for all the developers that if any vendoring changes are introduced, it is the responsibility of the developer who is making these changes to update `dep`'s files (although this might be obvious to the folks regularly using `dep`, this is a reminder to all of us - you, @miha-stopar, @zitnik and myself - since I believe none of us used `dep` workflows before).